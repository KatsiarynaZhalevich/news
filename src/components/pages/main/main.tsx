
import { useEffect } from "react";
import Pagination from "../../common/pagination/pagination";
import { useSelector } from "react-redux";
import Search from "./search/search";
import { useDispatch } from "react-redux";
import { postsRequested } from "../../common/store/slices/fetchPostsSlice";
import getPostsData from "../../common/store/selectors/fetchPostsSelectors";
import getPostsSettings from "../../common/store/selectors/settingSelectors";
import NewsList from "./newsList/newsList";

const Main = () => {

  const settings = useSelector(getPostsSettings);
  const posts = useSelector(getPostsData);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(postsRequested());
  }, [settings]);

  return <div className="p-5 h-full">
              <Search />
              <NewsList posts={posts}/> 
              {(!posts.loading && posts.pageCount !== 0) ? <Pagination pageCount={posts.pageCount}/> : null}       
          </div>
    
}

export default Main
