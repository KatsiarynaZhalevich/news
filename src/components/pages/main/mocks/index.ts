
export const mockStore = {
  page: {
    currentPage: 1,
    searchType: '&query=',
    searchStr: ''
  },
  posts: {
    posts: [
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'nalgeon'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Go 1.22: Interactive release notes'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://antonz.org/go-1-22/'
          }
        },
        _tags: [
          'story',
          'author_nalgeon',
          'story_38877667'
        ],
        author: 'nalgeon',
        created_at: '2024-01-05T10:40:17Z',
        created_at_i: 1704451217,
        num_comments: 0,
        objectID: '38877667',
        points: 1,
        story_id: 38877667,
        title: 'Go 1.22: Interactive release notes',
        updated_at: '2024-01-05T10:40:33Z',
        url: 'https://antonz.org/go-1-22/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'annabyrd'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: '16-yr-old is shaking up heady, often orbit of World Darts Championship'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://www.cnn.com/2024/01/02/sport/luke-littler-world-darts-championship-alexandra-palace-spt-intl/index.html'
          }
        },
        _tags: [
          'story',
          'author_annabyrd',
          'story_38877637'
        ],
        author: 'annabyrd',
        children: [
          38877654
        ],
        created_at: '2024-01-05T10:37:28Z',
        created_at_i: 1704451048,
        num_comments: 1,
        objectID: '38877637',
        points: 1,
        story_id: 38877637,
        title: '16-yr-old is shaking up heady, often orbit of World Darts Championship',
        updated_at: '2024-01-05T10:40:33Z',
        url: 'https://www.cnn.com/2024/01/02/sport/luke-littler-world-darts-championship-alexandra-palace-spt-intl/index.html'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'cannibalXxx'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Do you like programming? Then come and learn with me here for free'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://chat-to.dev/chat?q=python-solutions'
          }
        },
        _tags: [
          'story',
          'author_cannibalXxx',
          'story_38877619'
        ],
        author: 'cannibalXxx',
        created_at: '2024-01-05T10:35:12Z',
        created_at_i: 1704450912,
        num_comments: 0,
        objectID: '38877619',
        points: 1,
        story_id: 38877619,
        title: 'Do you like programming? Then come and learn with me here for free',
        updated_at: '2024-01-05T10:37:33Z',
        url: 'https://chat-to.dev/chat?q=python-solutions'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'thinkingemote'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Charity for Profit: Company Made Millions Off Twitch Fundraising Streams'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://www.jacobwolf.report/p/charity-profit-look-company-made-millions-off-twitchs-mostwatched-fundraising-streams'
          }
        },
        _tags: [
          'story',
          'author_thinkingemote',
          'story_38877616'
        ],
        author: 'thinkingemote',
        created_at: '2024-01-05T10:34:48Z',
        created_at_i: 1704450888,
        num_comments: 0,
        objectID: '38877616',
        points: 1,
        story_id: 38877616,
        title: 'Charity for Profit: Company Made Millions Off Twitch Fundraising Streams',
        updated_at: '2024-01-05T10:37:33Z',
        url: 'https://www.jacobwolf.report/p/charity-profit-look-company-made-millions-off-twitchs-mostwatched-fundraising-streams'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'laserstrahl'
          },
          story_text: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Are there any mixing programs for Windows you can recommend?\nI installed rekordbox but it comes with a paid plan.\nPreferably open source.\nEasy intuitive if possible.'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Ask HN: Looking for a DJ Program'
          }
        },
        _tags: [
          'story',
          'author_laserstrahl',
          'story_38877604',
          'ask_hn'
        ],
        author: 'laserstrahl',
        created_at: '2024-01-05T10:33:32Z',
        created_at_i: 1704450812,
        num_comments: 0,
        objectID: '38877604',
        points: 1,
        story_id: 38877604,
        story_text: 'Are there any mixing programs for Windows you can recommend?\nI installed rekordbox but it comes with a paid plan.\nPreferably open source.\nEasy intuitive if possible.',
        title: 'Ask HN: Looking for a DJ Program',
        updated_at: '2024-01-05T10:37:33Z'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'w3ll_w3ll_w3ll'
          },
          story_text: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'MO704022, Last updated: January 5, 2024 at 4:18 AM GMT+1<p>Title: One percent of Google Chrome users may experience degraded functionality when accessing Microsoft 365 services<p>User impact: Users may experience degraded functionality when accessing Microsoft 365 services via Google Chrome.<p>More info: While this may not be a viable solution for your organization, users can leverage Microsoft Edge to work around the issue while we develop mitigations.<p>At this time, we\'ve confirmed the following Microsoft 365 impact scenarios when accessing via the Google Chrome web browser:<p>SharePoint Online\n- Unable to create sites from SharePoint or the admin center\n- Unable to access Viva Engage or connector webparts\nOneDrive for Business\n- Unable to access recent documents via the Microsoft 365 app launcher\nMicrosoft Teams\n- Unable to access OneDrive from the app tray\n- Unable to access the Files tab within a Teams channel\n- Unable to access Viva connections<p>Current status: We\'re validating additional impact scenarios to Microsoft 365 services and will continue to update the &quot;More Info&quot; section of this communication as that work progresses. In parallel, we are developing and deploying changes to restore functionality when accessing via the affected web browser.<p>Scope of impact: Currently one percent of Google Chrome users may experience impact.<p>Start time: Thursday, January 4, 2024 at 1:00 AM GMT+1<p>Root cause: A recent third-party network change is preventing some Microsoft 365 features to function as expected when accessed via a specific web browser.<p>Next update by: Saturday, January 6, 2024 at 4:00 AM GMT+1'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'One percent of Chrome users experience degraded functionality on Microsoft 365'
          }
        },
        _tags: [
          'story',
          'author_w3ll_w3ll_w3ll',
          'story_38877588',
          'ask_hn'
        ],
        author: 'w3ll_w3ll_w3ll',
        created_at: '2024-01-05T10:31:14Z',
        created_at_i: 1704450674,
        num_comments: 0,
        objectID: '38877588',
        points: 1,
        story_id: 38877588,
        story_text: 'MO704022, Last updated: January 5, 2024 at 4:18 AM GMT+1<p>Title: One percent of Google Chrome users may experience degraded functionality when accessing Microsoft 365 services<p>User impact: Users may experience degraded functionality when accessing Microsoft 365 services via Google Chrome.<p>More info: While this may not be a viable solution for your organization, users can leverage Microsoft Edge to work around the issue while we develop mitigations.<p>At this time, we&#x27;ve confirmed the following Microsoft 365 impact scenarios when accessing via the Google Chrome web browser:<p>SharePoint Online\n- Unable to create sites from SharePoint or the admin center\n- Unable to access Viva Engage or connector webparts\nOneDrive for Business\n- Unable to access recent documents via the Microsoft 365 app launcher\nMicrosoft Teams\n- Unable to access OneDrive from the app tray\n- Unable to access the Files tab within a Teams channel\n- Unable to access Viva connections<p>Current status: We&#x27;re validating additional impact scenarios to Microsoft 365 services and will continue to update the &quot;More Info&quot; section of this communication as that work progresses. In parallel, we are developing and deploying changes to restore functionality when accessing via the affected web browser.<p>Scope of impact: Currently one percent of Google Chrome users may experience impact.<p>Start time: Thursday, January 4, 2024 at 1:00 AM GMT+1<p>Root cause: A recent third-party network change is preventing some Microsoft 365 features to function as expected when accessed via a specific web browser.<p>Next update by: Saturday, January 6, 2024 at 4:00 AM GMT+1',
        title: 'One percent of Chrome users experience degraded functionality on Microsoft 365',
        updated_at: '2024-01-05T10:32:18Z'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'straumat'
          },
          story_text: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Introducing Royllo Explorer: an open source search engine / explorer / API for Taproot Assets living on the Bitcoin blockchain. Think of it as a &quot;Google&quot; for Taproot Assets, providing you with a web interface and an API to search and discover these assets with ease.<p>With this release, we now store and display assets meta'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Show HN: Royllo 0.1.0, an open source explorer for Taproot assets'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://github.com/royllo/explorer'
          }
        },
        _tags: [
          'story',
          'author_straumat',
          'story_38877587',
          'show_hn'
        ],
        author: 'straumat',
        created_at: '2024-01-05T10:31:07Z',
        created_at_i: 1704450667,
        num_comments: 0,
        objectID: '38877587',
        points: 1,
        story_id: 38877587,
        story_text: 'Introducing Royllo Explorer: an open source search engine &#x2F; explorer &#x2F; API for Taproot Assets living on the Bitcoin blockchain. Think of it as a &quot;Google&quot; for Taproot Assets, providing you with a web interface and an API to search and discover these assets with ease.<p>With this release, we now store and display assets meta',
        title: 'Show HN: Royllo 0.1.0, an open source explorer for Taproot assets',
        updated_at: '2024-01-05T10:32:18Z',
        url: 'https://github.com/royllo/explorer'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'thunderbong'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Inuits and Kayaks in Aberdeen'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://eastpolepaddles.com/inuit-kayak/'
          }
        },
        _tags: [
          'story',
          'author_thunderbong',
          'story_38877580'
        ],
        author: 'thunderbong',
        created_at: '2024-01-05T10:29:26Z',
        created_at_i: 1704450566,
        num_comments: 0,
        objectID: '38877580',
        points: 1,
        story_id: 38877580,
        title: 'Inuits and Kayaks in Aberdeen',
        updated_at: '2024-01-05T10:32:18Z',
        url: 'https://eastpolepaddles.com/inuit-kayak/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'thunderbong'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'What\'s the State of U.S. Procurement? Any Improvements in the Works?'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://old.reddit.com/r/CredibleDefense/comments/18wr57g/whats_the_state_of_us_procurement_any/'
          }
        },
        _tags: [
          'story',
          'author_thunderbong',
          'story_38877571'
        ],
        author: 'thunderbong',
        created_at: '2024-01-05T10:27:11Z',
        created_at_i: 1704450431,
        num_comments: 0,
        objectID: '38877571',
        points: 1,
        story_id: 38877571,
        title: 'What\'s the State of U.S. Procurement? Any Improvements in the Works?',
        updated_at: '2024-01-05T10:32:18Z',
        url: 'https://old.reddit.com/r/CredibleDefense/comments/18wr57g/whats_the_state_of_us_procurement_any/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'korh0b'
          },
          story_text: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Hi!<p>Stolbitsa is an ear training app that can use real music for exercises, not just randomly picked notes like all other similar applications. This is a GitHub OAuth app and every ear training lesson is a public GitHub repository. You can create your own lessons and publish them on Github to share with other players.<p>All lessons are divided into 12 difficulty levels equal to the amount of notes in chromatic scale. You start learning from level 1 and gradually increase difficulty until you become an expert.<p>If you pass a lesson correctly you are rewarded with silver notes. If you guess 90% of notes you get 1 silver note, 95% - 2, 100% - 3. Later you can use the earned silver notes to give them to the lessons you like. The more silver notes a lesson gets the higher it is displayed on the main page listing. You can also star a lesson on Github and leave a comment. These are social network features of the app that all other similar apps are lacking as well.<p>This is intended to be some kind of a social network of people interested in music education on top of Github. You create and publish your lessons and get comments and stars for them. Every user has a public profile where their achievements and lessons they created are listed.<p>Currently only several initial lessons are created for each difficulty level. But nevertheless the app can already be used as a fully functional ear trainer. New lessons are constantly being added and I hope that more will be created by users.<p>You can play directly in the browser or as PWA on any platform, also Android apk is available for download. Playing is better on mobile, editing - on desktop. To create a new lesson click the edit icon in any existing lesson, then edit it and publish it as yours. I think the editor is self-explanatory. After you have created a lesson it is not shown on the main page until you give it at least one silver note, you can only view it on your profile page.<p>Please tell what you think about it, play it and try to create your own lessons!'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Show HN: Stolbitsa – social ear training on GitHub'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://stolbitsa.com/'
          }
        },
        _tags: [
          'story',
          'author_korh0b',
          'story_38877565',
          'show_hn'
        ],
        author: 'korh0b',
        created_at: '2024-01-05T10:26:27Z',
        created_at_i: 1704450387,
        num_comments: 0,
        objectID: '38877565',
        points: 2,
        story_id: 38877565,
        story_text: 'Hi!<p>Stolbitsa is an ear training app that can use real music for exercises, not just randomly picked notes like all other similar applications. This is a GitHub OAuth app and every ear training lesson is a public GitHub repository. You can create your own lessons and publish them on Github to share with other players.<p>All lessons are divided into 12 difficulty levels equal to the amount of notes in chromatic scale. You start learning from level 1 and gradually increase difficulty until you become an expert.<p>If you pass a lesson correctly you are rewarded with silver notes. If you guess 90% of notes you get 1 silver note, 95% - 2, 100% - 3. Later you can use the earned silver notes to give them to the lessons you like. The more silver notes a lesson gets the higher it is displayed on the main page listing. You can also star a lesson on Github and leave a comment. These are social network features of the app that all other similar apps are lacking as well.<p>This is intended to be some kind of a social network of people interested in music education on top of Github. You create and publish your lessons and get comments and stars for them. Every user has a public profile where their achievements and lessons they created are listed.<p>Currently only several initial lessons are created for each difficulty level. But nevertheless the app can already be used as a fully functional ear trainer. New lessons are constantly being added and I hope that more will be created by users.<p>You can play directly in the browser or as PWA on any platform, also Android apk is available for download. Playing is better on mobile, editing - on desktop. To create a new lesson click the edit icon in any existing lesson, then edit it and publish it as yours. I think the editor is self-explanatory. After you have created a lesson it is not shown on the main page until you give it at least one silver note, you can only view it on your profile page.<p>Please tell what you think about it, play it and try to create your own lessons!',
        title: 'Show HN: Stolbitsa – social ear training on GitHub',
        updated_at: '2024-01-05T10:30:03Z',
        url: 'https://stolbitsa.com/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'ssmew'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Platform Engineering Still Run by Spreadsheet'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://thenewstack.io/year-in-review-platform-engineering-still-run-by-spreadsheet/'
          }
        },
        _tags: [
          'story',
          'author_ssmew',
          'story_38877550'
        ],
        author: 'ssmew',
        created_at: '2024-01-05T10:24:00Z',
        created_at_i: 1704450240,
        num_comments: 0,
        objectID: '38877550',
        points: 1,
        story_id: 38877550,
        title: 'Platform Engineering Still Run by Spreadsheet',
        updated_at: '2024-01-05T10:27:03Z',
        url: 'https://thenewstack.io/year-in-review-platform-engineering-still-run-by-spreadsheet/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'rakshran'
          },
          story_text: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'A 2 or 4-week development cycle where requirements are frozen might prevent the teams from solving for low-hanging opportunities identified mid-sprint that might significantly impact acquisitions/revenue. Are the best in class tech teams rigid about this or is the culture more flexible.'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Do high growth Silicon Valley startups (OpenAI, Stripe etc.) use agile?'
          }
        },
        _tags: [
          'story',
          'author_rakshran',
          'story_38877546',
          'ask_hn'
        ],
        author: 'rakshran',
        children: [
          38877597
        ],
        created_at: '2024-01-05T10:23:19Z',
        created_at_i: 1704450199,
        num_comments: 1,
        objectID: '38877546',
        points: 1,
        story_id: 38877546,
        story_text: 'A 2 or 4-week development cycle where requirements are frozen might prevent the teams from solving for low-hanging opportunities identified mid-sprint that might significantly impact acquisitions&#x2F;revenue. Are the best in class tech teams rigid about this or is the culture more flexible.',
        title: 'Do high growth Silicon Valley startups (OpenAI, Stripe etc.) use agile?',
        updated_at: '2024-01-05T10:33:18Z'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'rrampage'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'The Life of a Data Byte'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://blog.jessfraz.com/post/the-life-of-a-data-byte/'
          }
        },
        _tags: [
          'story',
          'author_rrampage',
          'story_38877537'
        ],
        author: 'rrampage',
        created_at: '2024-01-05T10:22:20Z',
        created_at_i: 1704450140,
        num_comments: 0,
        objectID: '38877537',
        points: 1,
        story_id: 38877537,
        title: 'The Life of a Data Byte',
        updated_at: '2024-01-05T10:27:03Z',
        url: 'https://blog.jessfraz.com/post/the-life-of-a-data-byte/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'FredrikMeyer'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'How I set up my new MacBook'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://fredrikmeyer.net//2024/01/05/new-mac.html'
          }
        },
        _tags: [
          'story',
          'author_FredrikMeyer',
          'story_38877484'
        ],
        author: 'FredrikMeyer',
        children: [
          38877485
        ],
        created_at: '2024-01-05T10:14:25Z',
        created_at_i: 1704449665,
        num_comments: 1,
        objectID: '38877484',
        points: 2,
        story_id: 38877484,
        title: 'How I set up my new MacBook',
        updated_at: '2024-01-05T10:37:33Z',
        url: 'https://fredrikmeyer.net//2024/01/05/new-mac.html'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'iacguy'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'How to Become a Hacker'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'http://www.catb.org/~esr/faqs/hacker-howto.html'
          }
        },
        _tags: [
          'story',
          'author_iacguy',
          'story_38877463'
        ],
        author: 'iacguy',
        created_at: '2024-01-05T10:10:27Z',
        created_at_i: 1704449427,
        num_comments: 0,
        objectID: '38877463',
        points: 2,
        story_id: 38877463,
        title: 'How to Become a Hacker',
        updated_at: '2024-01-05T10:22:03Z',
        url: 'http://www.catb.org/~esr/faqs/hacker-howto.html'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'ustad'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Open source collection of Google\'s C++ libraries'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://abseil.io/'
          }
        },
        _tags: [
          'story',
          'author_ustad',
          'story_38877460'
        ],
        author: 'ustad',
        created_at: '2024-01-05T10:09:35Z',
        created_at_i: 1704449375,
        num_comments: 0,
        objectID: '38877460',
        points: 1,
        story_id: 38877460,
        title: 'Open source collection of Google\'s C++ libraries',
        updated_at: '2024-01-05T10:11:18Z',
        url: 'https://abseil.io/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'mcint'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'iPhone keyboard: Learns what you type, dynamically resizes key "hot zones"(2008)'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://www.cnet.com/tech/mobile/iphone-keyboard-learns-what-you-type-dynamically-resizes-key-hot-zones/'
          }
        },
        _tags: [
          'story',
          'author_mcint',
          'story_38877456'
        ],
        author: 'mcint',
        children: [
          38877457
        ],
        created_at: '2024-01-05T10:08:36Z',
        created_at_i: 1704449316,
        num_comments: 1,
        objectID: '38877456',
        points: 1,
        story_id: 38877456,
        title: 'iPhone keyboard: Learns what you type, dynamically resizes key "hot zones"(2008)',
        updated_at: '2024-01-05T10:11:18Z',
        url: 'https://www.cnet.com/tech/mobile/iphone-keyboard-learns-what-you-type-dynamically-resizes-key-hot-zones/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'signa11'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'A look at terminal emulators, part 1 (2018)'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://lwn.net/Articles/749992/'
          }
        },
        _tags: [
          'story',
          'author_signa11',
          'story_38877455'
        ],
        author: 'signa11',
        created_at: '2024-01-05T10:08:22Z',
        created_at_i: 1704449302,
        num_comments: 0,
        objectID: '38877455',
        points: 1,
        story_id: 38877455,
        title: 'A look at terminal emulators, part 1 (2018)',
        updated_at: '2024-01-05T10:11:18Z',
        url: 'https://lwn.net/Articles/749992/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'geox'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'Tokyo\'s top tuna for the new year sells for ¥114.2M ($788,440)'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://www.japantimes.co.jp/business/2024/01/05/tuna-new-year-auction/'
          }
        },
        _tags: [
          'story',
          'author_geox',
          'story_38877454'
        ],
        author: 'geox',
        created_at: '2024-01-05T10:08:20Z',
        created_at_i: 1704449300,
        num_comments: 0,
        objectID: '38877454',
        points: 3,
        story_id: 38877454,
        title: 'Tokyo\'s top tuna for the new year sells for ¥114.2M ($788,440)',
        updated_at: '2024-01-05T10:32:18Z',
        url: 'https://www.japantimes.co.jp/business/2024/01/05/tuna-new-year-auction/'
      },
      {
        _highlightResult: {
          author: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'thefilmore'
          },
          title: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'A modest proposal for better errno values'
          },
          url: {
            matchLevel: 'none',
            matchedWords: [],
            value: 'https://groups.google.com/g/fa.freebsd.hackers/c/V7cWmQeWVfg/m/IIjCCbdz4BEJ'
          }
        },
        _tags: [
          'story',
          'author_thefilmore',
          'story_38877442'
        ],
        author: 'thefilmore',
        created_at: '2024-01-05T10:06:00Z',
        created_at_i: 1704449160,
        num_comments: 0,
        objectID: '38877442',
        points: 1,
        story_id: 38877442,
        title: 'A modest proposal for better errno values',
        updated_at: '2024-01-05T10:11:18Z',
        url: 'https://groups.google.com/g/fa.freebsd.hackers/c/V7cWmQeWVfg/m/IIjCCbdz4BEJ'
      }
    ],
    loading: false,
    errors: '',
    fetched: true,
    pageCount: 50
  }
}