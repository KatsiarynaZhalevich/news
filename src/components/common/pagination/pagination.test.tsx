import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mockStore } from '../../pages/main/mocks';
import { render } from '@testing-library/react';
import Pagination from './pagination';

const mockStoreTest = configureStore([]);

const store = mockStoreTest(mockStore);

test('Pagination component should renders correctly', () => {
  const { asFragment } = render (
      <Provider store={store}>
        <Pagination pageCount={4} />
      </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
})