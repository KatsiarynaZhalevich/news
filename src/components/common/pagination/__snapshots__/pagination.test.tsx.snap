// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="prevPage"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowLeftRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.71 15.88 10.83 12l3.88-3.88c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.39.39-1.03 0-1.42z"
        />
      </svg>
    </button>
    <span
      class="inline-block mx-4"
      data-testid="pages"
    >
      1 of 4
    </span>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="nextPage"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowRightRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.29 15.88 13.17 12 9.29 8.12a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0l4.59 4.59c.39.39.39 1.02 0 1.41L10.7 17.3c-.39.39-1.02.39-1.41 0-.38-.39-.39-1.03 0-1.42z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
